addendumer.go

```diff
@@ -12,3 +17,149 @@ type Addendumer interface {
 	String() string
 	Validate() error
 }
+
+type addenda struct {
+	ID string `json:"id"` // EntryDetail.ID that this addenda belongs to
+
+	Addenda Addendumer
+}
+
+// UnmarshalJSON attempts to parse data as each AddendaXX record.
+func (a *addenda) UnmarshalJSON(data []byte) error {
+	addenda02 := NewAddenda02()
```

Comment:
I thought about codegen for this method, thoughts?

Reply:
Give me a sample, Code Gen in legacy never worked and the clients would all scrap it and write their own code.

Reply:
It would just be in our codebase to create that `UnmarshalJSON` method. Read all addenda types from the source tree and write each `if err := json.NewDecoder(...)...`